src/main/webapp/projectReview/ProjectReview.js

```diff
@@ -330,19 +354,38 @@ class ProjectReview extends Component {
     if (fundings !== null && fundings.length > 0) {
       fundings.map((f, idx) => {
         let funding = {};
-        funding.source = f.source.label;
-        funding.awardNumber = f.identifier;
-        funding.name = f.sponsor;
-        fundingList.push(funding);
+        if (!this.isEmpty(f.future.source.label)) {
+          funding.source = f.future.source.label;
+          funding.award = f.future.identifier;
+          funding.name = f.future.sponsor;
+          fundingList.push(funding);
+        }
       });
     }
     return fundingList;
   }
 
   compareObj(obj1, obj2) {
-    return JSON.stringify(this.state[obj1]) === JSON.stringify(this.state[obj2]);
+    let form1 = JSON.parse(JSON.stringify(this.state[obj1]));
+    let form2 = JSON.parse(JSON.stringify(this.state[obj2]));
+    if ( form1 !== undefined && form2 !== undefined && form1.fundings !== undefined && form2.fundings !== undefined) {
```

Comment:
Could you do something like this instead? (https://www.w3schools.com/jsref/jsref_every.asp)
```
[form1, form2, form1.fundings, form2.fundings].every(function(el){ return el !== undefined })
```

Reply:
Alternatively, we could use lodash.get which has a form of null-safe traversal: https://lodash.com/docs/#get

Reply:
I admit to being spoiled by Groovy's null-safe handling - it's one of the things that keeps me coming back to languages like it and Kotlin.

Reply:
I really liked lodash approach, so I've imported it to the project. Thanks for the suggestion! now we can count on this library.